### ISSUE TITLE
disregarding vote

### ISSUE THREAD

**Issue by x1p** [BenCodez/VotingPlugin#614]
I have a new player. Comes online, plays some, then leaves.
After an hour or so, he votes. Votes arrive but are disregarded
```
Recieved a vote from service site 'votesite1' by player 'dirkahles'!
Player dirkahles has not joined before, disregarding vote, set AllowUnjoined to true to prevent this
```
I can't give him votes, the system ignores him:
```
Gave dirkahles 2 points, dirkahles now has 0 points
Player does not exist: dirkahles
```
I have never ever seen this behavior before. Is there something changed, or did I miss something?
I would expect, that when a player has been online, votes should be accepted by the system.

VotingPlugin 5.19.4

**Comment by BenCodez** [comment]
Server version? Set allowunjoined to true?

**Comment by x1p** [comment]
1.13.2
AllowUnjoined: false
I have set AllowUnjoined to false because I don't want:
- fake names to spam in the public chat
- pollute the database

It's probable solved after a reboot of the server, but I think it should work instant

**Comment by BenCodez** [comment]
is check name mojang enabled?

**Comment by BenCodez** [comment]
http://ben12345rocks.com/job/VotingPlugin/

That should help also.

Edit: Make sure its build 306

**Comment by x1p** [comment]
Sorry for the late reply.
CheckNameMojang is deprecated so that is not available anymore

I'll try as soon as possible the latest build
I trust that this is solved. So I'll close it for now to keep things tidy.

**Comment by BenCodez** [comment]
Make sure it's the last build, I made more improvements to it.

**Comment by x1p** [comment]
No problem

**Comment by x1p** [comment]
Unfortunately it happened again.
```
[13:26:27 INFO]: UUID of player DanielSzy is d0825917-05d0-4228-a661-9b7826830546
[13:26:27 INFO]: DanielSzy[/xxx.xxx.xxx.xxx] logged in with entity id 61 at ([world]-1.0, 66.0, 1.6)
[13:26:36 INFO]: DanielSzy lost connection: Disconnected
...
[13:32:18 INFO]: [VotingPlugin] Recieved a vote from service site 'votesite1' by player 'DanielZsy'!
[13:32:18 WARN]: [VotingPlugin] Player DanielZsy has not joined before, disregarding vote, set AllowUnjoined to true to prevent this
[13:35:50 INFO]: [VotingPlugin] Recieved a vote from service site 'votesite1' by player 'DanielSzy'!
```
I was able to give him the point manually.
In this case VotingPlugin 5.20.2 build 395 was used

**Comment by BenCodez** [comment]
This set to true?
AllowUnJoinedCheckServer: true

**Comment by x1p** [comment]
```
# Let players who never joined before vote
# Recommend: False (Will prevent creating random user data)
AllowUnjoined: false

# Use server lookup for when allowunjoined is set to false
# Recommend leaving enabled unless you face issues
AllowUnJoinedCheckServer: true
```

**Comment by BenCodez** [comment]
http://ben12345rocks.com/job/VotingPlugin/

This better?

**Comment by x1p** [comment]
I think build 413, commit 3e429b8 solves another issue, or maybe I'm too quick.

**Comment by BenCodez** [comment]
This fixed it I hope https://github.com/Ben12345rocks/AdvancedCore/commit/47714bc99ac6b8499296da1036bb041b15ff7599

**Comment by x1p** [comment]
Ah ok. I didn't see that one. I'll put it on the servers and will let you know if it happens again.

**Comment by x1p** [comment]
Is this related? I think it is.
A new player joins and this appeared:
```
[20:50:39 WARN]: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '453d800e-8df6-4687-b631-0547cdcd475b' for key 'PRIMARY'
[20:50:39 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[20:50:39 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[20:50:39 WARN]: at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[20:50:39 WARN]: at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[20:50:39 WARN]: at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[20:50:39 WARN]: at com.mysql.jdbc.Util.getInstance(Util.java:408)
[20:50:39 WARN]: at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
[20:50:39 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
[20:50:39 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
[20:50:39 WARN]: at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
[20:50:39 WARN]: at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
[20:50:39 WARN]: at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2495)
[20:50:39 WARN]: at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1903)
[20:50:39 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2124)
[20:50:39 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2058)
[20:50:39 WARN]: at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5158)
[20:50:39 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2043)
[20:50:39 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
[20:50:39 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
[20:50:39 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query.executeUpdate(Query.java:178)
[20:50:39 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query$3.run(Query.java:213)
[20:50:39 WARN]: at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[20:50:39 WARN]: at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[20:50:39 WARN]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[20:50:39 WARN]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[20:50:39 WARN]: at java.lang.Thread.run(Thread.java:748)
```

**Comment by BenCodez** [comment]
This better?
http://ben12345rocks.com/job/VotingPlugin/

**Comment by x1p** [comment]
I'm cautious to test further with newer versions beyond 415 on production servers because of issue #649

**Comment by x1p** [comment]
VotingPlugin 5.20.2 build 429 gives error on joining new player
```
[19:07:22 WARN]: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '28c85d2d-990b-44a9-80fa-b965e320db7a' for key 'PRIMARY'
[19:07:22 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[19:07:22 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[19:07:22 WARN]: at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[19:07:22 WARN]: at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[19:07:22 WARN]: at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[19:07:22 WARN]: at com.mysql.jdbc.Util.getInstance(Util.java:408)
[19:07:22 WARN]: at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
[19:07:22 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
[19:07:22 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
[19:07:22 WARN]: at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
[19:07:22 WARN]: at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
[19:07:22 WARN]: at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2495)
[19:07:22 WARN]: at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1903)
[19:07:22 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2124)
[19:07:22 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2058)
[19:07:22 WARN]: at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5158)
[19:07:22 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2043)
[19:07:22 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
[19:07:22 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
[19:07:22 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query.executeUpdate(Query.java:178)
[19:07:22 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query$3.run(Query.java:213)
[19:07:22 WARN]: at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[19:07:22 WARN]: at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[19:07:22 WARN]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[19:07:22 WARN]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[19:07:22 WARN]: at java.lang.Thread.run(Thread.java:748)
```

**Comment by BenCodez** [comment]
http://ben12345rocks.com/job/VotingPlugin/

Maybe now? Build #441

**Comment by x1p** [comment]
```
[21:14:51 WARN]: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '13388aff-f6ab-4ab9-b541-08da99273a6f' for key 'PRIMARY'
[21:14:51 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[21:14:51 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[21:14:51 WARN]: at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[21:14:51 WARN]: at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[21:14:51 WARN]: at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[21:14:51 WARN]: at com.mysql.jdbc.Util.getInstance(Util.java:408)
[21:14:51 WARN]: at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
[21:14:51 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
[21:14:51 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
[21:14:51 WARN]: at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
[21:14:51 WARN]: at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
[21:14:51 WARN]: at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2495)
[21:14:51 WARN]: at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1903)
[21:14:51 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2124)
[21:14:51 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2058)
[21:14:51 WARN]: at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5158)
[21:14:51 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2043)
[21:14:51 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
[21:14:51 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
[21:14:51 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query.executeUpdate(Query.java:178)
[21:14:51 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query$3.run(Query.java:213)
[21:14:51 WARN]: at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[21:14:51 WARN]: at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[21:14:51 WARN]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[21:14:51 WARN]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[21:14:51 WARN]: at java.lang.Thread.run(Thread.java:748)
```

**Comment by BenCodez** [comment]
Hoping this should do it:
http://ben12345rocks.com/job/VotingPlugin/

Build 444

**Comment by x1p** [comment]
Haven't seen it today. Let's hope it stays away.

I'll close this issue :+1:

**Comment by x1p** [comment]
I had a new player today. Voting went fine, message appeared.
However. Votes did not get saved in the database.
The placeholder for the featherboard said 0 points.
Then tried to retrieve points by executing /votepoints thatplayer
It returned that the player didn't exist (while he was online).
I gave him 2 points which seem to be accepted.

Votingplugin 5.20.3

**Comment by BenCodez** [comment]
http://ben12345rocks.com/job/VotingPlugin/

Any better?
If you had opened the vote gui or something it would of solve it. The plugin isn't automaticlly caching the data, so it needs something to force it to update. Hoping that should fix that, but can't really test it.

**Comment by x1p** [comment]
Jenkins has some problem. I can't download latest job right now

**Comment by BenCodez** [comment]
Having some VPS issues, working on getting it fixed. It's working now though.

**Comment by x1p** [comment]
I live in another time zone. So by the time I read your reply, hours have past.
It's down again now. I'll keep checking.

**Comment by BenCodez** [comment]
I rebooted it, see if that helps

**Comment by x1p** [comment]
thnx

**Comment by x1p** [comment]
I haven't tested it out because I have to wait for the nightly reboot to happen. But there are more odd things with a player.
There is this player who newly joined 2 days ago.
He voted after about 10 minutes. Didn't see his placeholder with number of votes rise. I tried to do /votepoints thisplayer and the votingplugin said: The player doesn't exist. I gave him 2 points manually. This is the issue stated above.
Yesterday, he voted, bought a few things with his new points. Didn't here him complain.
Today, he voted again. Said he didn't receive 1 of 2 points. Bought something. When I try to find out his votepoints, votingplugin says the player still doesn't exist. I searched the mysql database manually and his name/uuid is not present.
I think I'm getting nuts.

**Comment by BenCodez** [comment]
You'll need to see if there is any errors. It might of failed to saved to mysql database, but I got no idea.

Did the player vote while online or offline?

I don't currently have a mysql database to test with since my vps is having some issues (waiting on my host)

http://ben12345rocks.com/job/VotingPlugin/

Added some debug for mysql caching to that ^

Set log debug to file on and it will log to that one file.

**Comment by x1p** [comment]
He voted while being  (18 min) online on the server the vote was received. By now, he is online for 2 days now so I should expect the cache to be written to the database. I think the problem probably lies in the fact that I gave him points manually although he didn't was recognized as a player when he first joined. I also reloaded Votingplugin in order to have it push it's cache to the database. No error was seen from Votingplugin that day or when the server shut down.

I've put debug on INFO and will keep an eye on it.

**Comment by x1p** [comment]
I have to make a side note that it's a bit annoying to see all debug messages on the console despite:
```
# Will log debug messages to VotingPlugin/Log/log.txt
# Only needed for extreme cases
LogDebugToFile: true
```
It would be nice if logging to a file is true, the console would silence :-)

**Comment by x1p** [comment]
[Debug.txt](https://github.com/Ben12345rocks/VotingPlugin/files/3771617/Debug.txt)
Attached is a file with debug output of 2 votes receiving.
First one is the problematic one, second vote is from a normal player with no problems.

**Comment by BenCodez** [comment]
It should log without having debug set. So only logging to the file. Yeah, I still can't figure out exactly what is going on.

**Comment by x1p** [comment]
Can I add this player manually with some command or something?

**Comment by BenCodez** [comment]
Just add it manually to the database and /av clearcache? Or force a vote? Are you sure it doesn't exist?

**Comment by x1p** [comment]
I'm 100% sure he doesn't exist. I've searched the database by name and uuid multiple times.
Forcevote makes a perfect broadcast but still doesn't add him to the database.

I've added him manually and it seems to work. I can check his votes in game.

**Comment by BenCodez** [comment]
My guess it failed to send the query to add him, but the plugin thought it worked. You can set clearcacheonjoin. Is there more than 1 server with the same database?

**Comment by x1p** [comment]
Yes there are 3 servers connected

**Comment by BenCodez** [comment]
All clear cache options on are right?

**Comment by x1p** [comment]
```
# Clear cache on update, setting to true reduces update speed
ClearCacheOnUpdate: true

# Clear mysql on player join always
ClearCacheOnJoin: true

# Clears cache for the player who voted only
# Only effects if using mysql
# Would recommend for some bungee setups
ClearCacheOnVote: true

# Will force GUI's to close on every click
AlwaysCloseInventory: false
```

**Comment by BenCodez** [comment]
Hmm, I'll need to do some testing.

**Comment by x1p** [comment]
What I saw was: the player comes online, I do /votepoints to check his data, and the plugin still says the player doesn't exist. How odd when he is online?

**Comment by BenCodez** [comment]
I got no idea right now.

**Comment by x1p** [comment]
Meanwhile, more player names are emerging who don't exist

**Comment by BenCodez** [comment]
There is a database issue in 5.20.3 where new users sometimes don't get created.
Hoping this fixes it:
http://ben12345rocks.com/job/VotingPlugin/

**Comment by x1p** [comment]
The votes seem to work fine for new players :ok_hand: 
Only this error appeared:
```
[15:50:45 WARN]: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry 'eb680d6d-eb1e-4561-8f33-f063877186b1' for key 'PRIMARY'
[15:50:45 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[15:50:45 WARN]: at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[15:50:45 WARN]: at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[15:50:45 WARN]: at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[15:50:45 WARN]: at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[15:50:45 WARN]: at com.mysql.jdbc.Util.getInstance(Util.java:408)
[15:50:45 WARN]: at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:936)
[15:50:45 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3978)
[15:50:45 WARN]: at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3914)
[15:50:45 WARN]: at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2530)
[15:50:45 WARN]: at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2683)
[15:50:45 WARN]: at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2495)
[15:50:45 WARN]: at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1903)
[15:50:45 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2124)
[15:50:45 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2058)
[15:50:45 WARN]: at com.mysql.jdbc.PreparedStatement.executeLargeUpdate(PreparedStatement.java:5158)
[15:50:45 WARN]: at com.mysql.jdbc.PreparedStatement.executeUpdate(PreparedStatement.java:2043)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.api.queries.Query.executeUpdate(Query.java:178)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.MySQL.insertQuery(MySQL.java:457)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.MySQL.insert(MySQL.java:443)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserStorage.mysql.MySQL.update(MySQL.java:548)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserManager.UserData.setString(UserData.java:267)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.UserManager.UserData.setStringList(UserData.java:283)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.Objects.User.setOfflineVotes(User.java:890)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.Objects.User.offVote(User.java:724)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.Listeners.PlayerJoinEvent.onPlayerLogin(PlayerJoinEvent.java:64)
[15:50:45 WARN]: at com.destroystokyo.paper.event.executor.asm.generated.GeneratedEventExecutor139.execute(Unknown Source)
[15:50:45 WARN]: at org.bukkit.plugin.EventExecutor.lambda$create$1(EventExecutor.java:69)
[15:50:45 WARN]: at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:76)
[15:50:45 WARN]: at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:66)
[15:50:45 WARN]: at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:520)
[15:50:45 WARN]: at com.Ben12345rocks.VotingPlugin.AdvancedCore.Listeners.PlayerJoinEvent$1.run(PlayerJoinEvent.java:85)
[15:50:45 WARN]: at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftTask.run(CraftTask.java:84)
[15:50:45 WARN]: at org.bukkit.craftbukkit.v1_13_R2.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:52)
[15:50:45 WARN]: at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22)
[15:50:45 WARN]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[15:50:45 WARN]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```

**Comment by BenCodez** [comment]
Should be okay with that error. Not sure how that happened right now.

**Comment by x1p** [comment]
It happens with every new player but ok fine, I'll try to ignore it.
Ah I know where it comes from. When a new player enters the lobby, this server adds the player correctly to the database. The other servers behind it try it too causing the warning.
Maybe there is a way to catch the exception and replace it with something like: `player already exists, nothing to add now.`

**Comment by BenCodez** [comment]
http://ben12345rocks.com/job/VotingPlugin/

This should help alot.

**Comment by x1p** [comment]
Works like a charm
